src/client_side.cc

```diff
@@ -4103,13 +4122,22 @@ ConnStateData::checkLogging()
 bool
 ConnStateData::mayTunnelUnsupportedProto()
 {
-    return Config.accessList.on_unsupported_protocol
+    if (!Config.accessList.on_unsupported_protocol)
+        return false;
+
 #if USE_OPENSSL
-           &&
-           ((port->flags.isIntercepted() && port->flags.tunnelSslBumping)
-            || (serverBump() && pinning.serverConnection))
+    // SslBump processing:
```

Comment:
The `1)` and `2)` items below can be misinterpreted as two "steps" rather than different "states" (handled by the same `if` statement).

```suggestion
    // Tunneling may be possible when parsing the TLS client handshake
    // and when parsing the first HTTP request on a bumped connection.
```

Reply:
I am resolving this change request although I am not sure that the "when A or when B" comment actually matches the "X && Y" code. I will ask you to clarify why there is a match in the next review.